src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -306,6 +310,56 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 ins = INS_crc32cx;
                 break;
 
+            case NI_AdvSimd_AddWideningLower:
+                assert(varTypeIsIntegral(intrin.baseType));
+                if (intrin.op1->TypeGet() == TYP_SIMD8)
+                {
+                    ins = varTypeIsUnsigned(intrin.baseType) ? INS_uaddl : INS_saddl;
```

Comment:
We should be able to get rid of this logic once we change the tables to take `simdSize` into account during lookup, correct?

Reply:
We can if we make simdSize and baseType of an intrinsic "independent" on each other - i.e. if we in addition to `HW_Flag_BaseTypeFromFirstArg`, `HW_Flag_BaseTypeFromSecondArg` we have `HW_Flag_SimdSizeFromFirstArg`, `HW_Flag_SimdSizeFromSecondArg` which would allow, for example, to set simdSize based on op1 and baseType from op2.

Then, the corresponding rows in hwintrinsiclistarm64.h could be updated as

```c++
HARDWARE_INTRINSIC(AdvSimd, AddWideningLower,  8, 2, {INS_saddl, INS_uaddl,    … HW_Flag_SimdSizeFromFirstArg|HW_Flag_BaseTypeFromSecondArg)
HARDWARE_INTRINSIC(AdvSimd, AddWideningLower, 16, 2, {INS_saddw, INS_uaddw,    … HW_Flag_SimdSizeFromFirstArg|HW_Flag_BaseTypeFromSecondArg)

HARDWARE_INTRINSIC(AdvSimd, AddWideningUpper,  8, 2, {INS_saddl2, INS_uaddl2,   … HW_Flag_SimdSizeFromFirstArg|HW_Flag_BaseTypeFromSecondArg)
HARDWARE_INTRINSIC(AdvSimd, AddWideningUpper, 16, 2, {INS_saddw2, INS_uaddw2,   … HW_Flag_SimdSizeFromFirstArg|HW_Flag_BaseTypeFromSecondArg)
```

In other words, if we allow *x and y coordinates* (where x is baseType and y is a tuple of (intrinsicId, simdSize)) of hwintrinsiclistarm64.h table to be orthogonal then the answer to your question is yes.